newlib/libc/time/mktime.c

```diff
@@ -79,19 +79,15 @@ ANSI C requires <<mktime>>.
 #include <time.h>
 #include "local.h"
 
-#define _SEC_IN_MINUTE 60L
-#define _SEC_IN_HOUR 3600L
-#define _SEC_IN_DAY 86400L
-
-static const int DAYS_IN_MONTH[12] =
+static const int8_t DAYS_IN_MONTH[12] =
 {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
 
 #define _DAYS_IN_MONTH(x) ((x == 1) ? days_in_feb : DAYS_IN_MONTH[x])
 
-static const int _DAYS_BEFORE_MONTH[12] =
+static const int16_t _DAYS_BEFORE_MONTH[12] =
```

Comment:
Would it be reasonable to save data space by computing this from DAYS_IN_MONTH?

Reply:
My guess is that the code to compute it will be slower and bigger than the array, but that may depend on the architecture.

Reply:
ah, that's a very good point. The array isn't very big ...